### ISSUE TITLE
A `CI/CD` validation check to Validate the Client SIde Queries with the talawa-api.

### ISSUE THREAD

**Issue by kb-0311** [PalisadoesFoundation/talawa-admin#923]
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is.

There have been many instances in the past where 

**Describe the solution you'd like**

This CI/CD check on a PR make sure that the the client side queries in talawa-admin will be consistent with the API's `schema.graphql` which is the complete schema definition in the API.

The check with the name `Runs Introspection on the github talawa-api repo on the schema.graphql file` validates the current state of the schema with the schema in the API and logs the breaking changes.

It is at this point where we can catch these errors in the CI/CD pipeline so that the admin project remains in sync with the API.

**Describe alternatives you've considered**

**Approach to be followed (optional)**
A clear and concise description of approach to be followed.

**Additional context**

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

**Comment by kb-0311** [comment]
This is the sample result of the above PR in my forked repository ->

https://github.com/kb-0311/talawa-admin/actions/runs/5229177481/jobs/9442012824?pr=1

Here is the format of how to inconsistencies will be relayed in the check.
![Screenshot from 2023-06-10 14-48-00](https://github.com/PalisadoesFoundation/talawa-admin/assets/96020697/b1b74a3a-2fbf-4ae3-809a-55ba548155e7)\

Here I had deliberately added an invalid query of `description` while querying `user`. Since, that does not exist in the API schema it logged the error.

**Comment by kb-0311** [comment]
@palisadoes pls assign this to me and please do create an repository secret called 
"INSPECTOR" and then create a github PAT with read-only access to public repository and assign its value to this "INSPECTOR" secret variable which will be used to fetch the schema of the talawa-api latest develop branch for the check to work.

**Comment by palisadoes** [comment]
Shouldn't we call this REPO_READ_ONLY or something similar so that the name is more generic? It could therefore be more easily understood based on its name, and could be applied to all the repos in the PalisadoesFoundation organization.

What do you think? Do you have any other suggestions?

I'm assuming that this is the only access it will require, correct? Please let me know.

![image](https://github.com/PalisadoesFoundation/talawa-admin/assets/16875803/494ae79d-9234-4013-91b4-7aa80f7176b3)

**Comment by kb-0311** [comment]
@palisadoes Yeah sure, you can give that secret 'REPO_READ_ONLY' name that is fine. Just that it has to be a classic Github Personal Access Token with only read-only access to public repos. 
After it is done,let me know I will update the PR to match this change.

**Comment by kb-0311** [comment]
@palisadoes Any updates on the secret variable?

**Comment by palisadoes** [comment]
It's complete. Please test

**Comment by kb-0311** [comment]
https://github.com/kb-0311/talawa-admin/actions/runs/5229177481/jobs/9442012824?pr=1

This is the same PR on my forked admin instance which returns results fine.

**Comment by palisadoes** [comment]
![image](https://github.com/PalisadoesFoundation/talawa-admin/assets/16875803/4941d6d7-e549-42a9-b431-3b934cd04f33)
The key is there in all the Talawa Repos

**Comment by kb-0311** [comment]
@palisadoes Ahh okay I see what the issue is, Basically PRs due to obvious security issues cannot access the secrets of parent repos.

Here is the result of the same PR in my forked repo where I had defined its PAT secret in the forked repo scope which does work.

https://github.com/kb-0311/talawa-admin/actions/runs/5229177481/jobs/9442012824?pr=1

I think it can be reviewed and merged, after merge, it will have access to all the secrets however to be on the safe side I have added this check as a push check too, so if it were to fail, we will know about it beforehand. 
I think reviewers should try out this check in their own forked repo.
